### PR TITLE
convert DATETIME columns when exporting from cache to CSV

### LINKED ISSUE #3704: [SQLLab] datetime columns interpreted as strings when exporting _cached_ result to CSV
### Foreword
I reported this issue in #3695, together with a related one.
As I found out that the two are not as closely related as I thought, I decided to split them...

### Superset version
0.20.4

### Expected results
Datetime columns are interpreted and formatted correcty when exporting a _cached_ query result as CSV.

### Actual results
The datetime columns are, when fetched from the cache, interpreted as string/object. When exporting the data to CSV, any `date_format` setting in the `CSV_EXPORT` options is without effect.

Even if no `date_format` is specified, the fromat of timestamps differd from the one obtained from CSV exports that are created without caching, as the default export fromat from pandas' `to_csv` method separates date and time with a space instead of a `T`. The latter is what is obtained when exporting the results from cache.

One major drawback of this issue is that MS Excel cannot _automatically_ deal with ISO timestamps which feature the `T` separator between date and time. Unfortunately, I think that we have to adapt to this embarassing inability of Excel.

### Steps to reproduce
Have a cache backend configured for superset (here: Redis) and execute and then export the following query:
```
select now()::timestamp as my_timestamp
```

### PR DESCRIPTION
**Pull request by rumbin**
Fixes #3704.

I'm not quite sure _why_ this is for the CSV export in SQLLab, while the CSV export in Explore handles DATETIME columns nicely when exporting from cached data.
To be honest, I could not manage to find the corresponding code used for Explore's CSV export. For this reason, this approach is totally independent.

### CODE REVIEW

**Comment by rumbin** [comment]
What am I supposed to do when the **coverage/coveralls** test fails?

**Comment by xrmx** [comment]
Add tests and while at it fixup whitespace in your code.

**Comment by rumbin** [comment]
@xrmx: I did not intruduce any whitespace. Could you please specify?

Furthermore I have no idea how to add a test and what needs to be tested and how. I am willing to contribute, but I'm really not an experienced programmer. Adding just two simple lines should not create so much overhead, in my opinion.

**Comment by rumbin** [comment]
Thinking about it: We use information that is at this point only present in the JSON stored in the cache. If the cache is not available — which I suppose for any generalized test —, then I really have no idea, how a test would have to be defined.

The whole `csv` function has no tests defined, as I understand. Maybe other parts of this function should be tested first?

@bkyryliuk I saw your name in the code; maybe you could give me a hand?

**Comment by Mogball** [comment]
If you just need to slightly bump coverage to be green and get some unit test writing experience, you can look at the coverage report for easy pickings.

https://github.com/rumbin/superset/blob/726e1a3abf0f0c98a55872497015be4bc9227c0e/superset/utils.py#L352

Might be a good place to start. Check out the other tests in `tests/utils_tests.py`

**Comment by bkyryliuk** [comment]
@rumbin - you should be able to mock the result backend, this way new lines will be triggered in the test.

Existing test lives in: https://github.com/apache/incubator-superset/blob/master/tests/core_tests.py#L554

**Comment by rumbin** [comment]
I don't want to sound stubborn, however, I do not understand why these two lines of code would require to write a test for the whole part of the function (exporting in case that a cache is configured) which was entirely untested beforehand.

The most relevant part of this functionality is not mine and the overhead of writing a test, especially the one of mocking a cache back end is way too much for my limited understanding of the code and the tiny contribution of this PR.

Please don't get me wrong. I do appreciate everybody's work on this wonderful piece of software and I do recognize that my contribution is mostly negligible. However, this is all that I can muster up at the moment.

In other words, I am afraid that I cannot provide the demanded test code. But it would be a bad decision to leave  issue #3704 unresolved for this reason, as it is a quite noticeable one for MS Excel end users.

My humble suggestion would be that the original author provides a test for the 'export from cache' functionality.

@mistercrunch: I'm not proud of this, but I would kindly ask you to take notice of this issue raised by the automatic coverage tests.
